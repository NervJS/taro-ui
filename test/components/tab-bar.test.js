import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtTabBar from '../../.temp/components/tab-bar/index'

describe('AtTabBar Snap', () => {
  const tabList = [
    { title: '待办事项', iconType: 'bullet-list', text: 'new' },
    { title: '拍照', iconType: 'camera' },
    { title: '文件夹', iconType: 'folder', text: '100', max: '99' },
    {
      title: '领取中心',
      image:
        'https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png',
      selectedImage:
        'https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png',
      text: 'new'
    }
  ]

  it('render initial AtTabBar', () => {
    const componet = renderToString(<AtTabBar />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabBar -- props customStyle', () => {
    const componet = renderToString(<AtTabBar customStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabBar -- props className', () => {
    const componet = renderToString(<AtTabBar className='test' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabBar -- props fixed', () => {
    const componet = renderToString(<AtTabBar fixed />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabBar -- props backgroundColor', () => {
    const componet = renderToString(<AtTabBar backgroundColor='red' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabBar -- props tabList', () => {
    const componet = renderToString(<AtTabBar tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabBar -- props current', () => {
    const componet = renderToString(<AtTabBar current='2' tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabBar -- props iconSize', () => {
    const componet = renderToString(
      <AtTabBar iconSize='26' tabList={tabList} />
    )
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabBar -- props fontSize', () => {
    const componet = renderToString(
      <AtTabBar fontSize='26' tabList={tabList} />
    )
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabBar -- props color', () => {
    const componet = renderToString(<AtTabBar color='red' tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabBar -- props selectedColor', () => {
    const componet = renderToString(
      <AtTabBar selectedColor='red' tabList={tabList} />
    )
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabBar -- props scroll', () => {
    const componet = renderToString(<AtTabBar scroll tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })
})
