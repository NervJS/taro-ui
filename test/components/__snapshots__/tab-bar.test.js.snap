// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTabBar Snap render AtTabBar -- props backgroundColor 1`] = `"<div style=\\"background-color:red;\\" class=\\"at-tab-bar\\"></div>"`;

exports[`AtTabBar Snap render AtTabBar -- props className 1`] = `"<div style=\\"background-color:;\\" class=\\"at-tab-bar test\\"></div>"`;

exports[`AtTabBar Snap render AtTabBar -- props color 1`] = `"<div style=\\"background-color:;\\" class=\\"at-tab-bar\\"><div style=\\"color:;\\" class=\\"at-tab-bar__item at-tab-bar__item--active\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-bullet-list\\"></span></div><div class=\\"at-badge__num\\">new</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">待办事项</div>undefined</div></div></div><div style=\\"color:red;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"color:red;font-size:;\\" class=\\"taro-text at-icon at-icon-camera\\"></span></div>undefined</div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">拍照</div>undefined</div></div></div><div style=\\"color:red;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"color:red;font-size:;\\" class=\\"taro-text at-icon at-icon-folder\\"></span></div><div class=\\"at-badge__num\\">99+</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">文件夹</div>undefined</div></div></div><div style=\\"color:red;\\" class=\\"at-tab-bar__item\\">undefined<div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img at-tab-bar__inner-img--inactive\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png\\"/></div><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png\\"/></div></div><div class=\\"at-badge__num\\">new</div></div><div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">领取中心</div>undefined</div></div></div></div>"`;

exports[`AtTabBar Snap render AtTabBar -- props current 1`] = `"<div style=\\"background-color:;\\" class=\\"at-tab-bar\\"><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-bullet-list\\"></span></div><div class=\\"at-badge__num\\">new</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">待办事项</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-camera\\"></span></div>undefined</div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">拍照</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-folder\\"></span></div><div class=\\"at-badge__num\\">99+</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">文件夹</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\">undefined<div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img at-tab-bar__inner-img--inactive\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png\\"/></div><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png\\"/></div></div><div class=\\"at-badge__num\\">new</div></div><div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">领取中心</div>undefined</div></div></div></div>"`;

exports[`AtTabBar Snap render AtTabBar -- props customStyle 1`] = `"<div style=\\"background-color:;color:red;\\" class=\\"at-tab-bar\\"></div>"`;

exports[`AtTabBar Snap render AtTabBar -- props fixed 1`] = `"<div style=\\"background-color:;\\" class=\\"at-tab-bar at-tab-bar--fixed\\"></div>"`;

exports[`AtTabBar Snap render AtTabBar -- props fontSize 1`] = `"<div style=\\"background-color:;\\" class=\\"at-tab-bar\\"><div style=\\"color:;\\" class=\\"at-tab-bar__item at-tab-bar__item--active\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-bullet-list\\"></span></div><div class=\\"at-badge__num\\">new</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:26px;\\" class=\\"at-tab-bar__title\\">待办事项</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-camera\\"></span></div>undefined</div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:26px;\\" class=\\"at-tab-bar__title\\">拍照</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-folder\\"></span></div><div class=\\"at-badge__num\\">99+</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:26px;\\" class=\\"at-tab-bar__title\\">文件夹</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\">undefined<div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img at-tab-bar__inner-img--inactive\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png\\"/></div><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png\\"/></div></div><div class=\\"at-badge__num\\">new</div></div><div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:26px;\\" class=\\"at-tab-bar__title\\">领取中心</div>undefined</div></div></div></div>"`;

exports[`AtTabBar Snap render AtTabBar -- props iconSize 1`] = `"<div style=\\"background-color:;\\" class=\\"at-tab-bar\\"><div style=\\"color:;\\" class=\\"at-tab-bar__item at-tab-bar__item--active\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:26px;\\" class=\\"taro-text at-icon at-icon-bullet-list\\"></span></div><div class=\\"at-badge__num\\">new</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">待办事项</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:26px;\\" class=\\"taro-text at-icon at-icon-camera\\"></span></div>undefined</div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">拍照</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:26px;\\" class=\\"taro-text at-icon at-icon-folder\\"></span></div><div class=\\"at-badge__num\\">99+</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">文件夹</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\">undefined<div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img at-tab-bar__inner-img--inactive\\" style=\\"width:26px;height:26px;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png\\"/></div><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img\\" style=\\"width:26px;height:26px;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png\\"/></div></div><div class=\\"at-badge__num\\">new</div></div><div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">领取中心</div>undefined</div></div></div></div>"`;

exports[`AtTabBar Snap render AtTabBar -- props scroll 1`] = `"<div style=\\"background-color:;\\" class=\\"at-tab-bar\\"><div style=\\"color:;\\" class=\\"at-tab-bar__item at-tab-bar__item--active\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-bullet-list\\"></span></div><div class=\\"at-badge__num\\">new</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">待办事项</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-camera\\"></span></div>undefined</div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">拍照</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-folder\\"></span></div><div class=\\"at-badge__num\\">99+</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">文件夹</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\">undefined<div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img at-tab-bar__inner-img--inactive\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png\\"/></div><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png\\"/></div></div><div class=\\"at-badge__num\\">new</div></div><div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">领取中心</div>undefined</div></div></div></div>"`;

exports[`AtTabBar Snap render AtTabBar -- props selectedColor 1`] = `"<div style=\\"background-color:;\\" class=\\"at-tab-bar\\"><div style=\\"color:red;\\" class=\\"at-tab-bar__item at-tab-bar__item--active\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"color:red;font-size:;\\" class=\\"taro-text at-icon at-icon-bullet-list\\"></span></div><div class=\\"at-badge__num\\">new</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">待办事项</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-camera\\"></span></div>undefined</div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">拍照</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-folder\\"></span></div><div class=\\"at-badge__num\\">99+</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">文件夹</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\">undefined<div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img at-tab-bar__inner-img--inactive\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png\\"/></div><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png\\"/></div></div><div class=\\"at-badge__num\\">new</div></div><div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">领取中心</div>undefined</div></div></div></div>"`;

exports[`AtTabBar Snap render AtTabBar -- props tabList 1`] = `"<div style=\\"background-color:;\\" class=\\"at-tab-bar\\"><div style=\\"color:;\\" class=\\"at-tab-bar__item at-tab-bar__item--active\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-bullet-list\\"></span></div><div class=\\"at-badge__num\\">new</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">待办事项</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-camera\\"></span></div>undefined</div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">拍照</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\"><div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-folder\\"></span></div><div class=\\"at-badge__num\\">99+</div></div>undefined<div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">文件夹</div>undefined</div></div></div><div style=\\"color:;\\" class=\\"at-tab-bar__item\\">undefined<div class=\\"at-badge\\"><div class=\\"at-tab-bar__icon\\"><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img at-tab-bar__inner-img--inactive\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png\\"/></div><div class=\\"taro-img taro-img__widthfix at-tab-bar__inner-img\\" style=\\"width:undefinedpx;height:undefinedpx;\\"><img class=\\"taro-img__mode-widthfix\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png\\"/></div></div><div class=\\"at-badge__num\\">new</div></div><div class=\\"\\"><div class=\\"at-badge\\"><div style=\\"font-size:;\\" class=\\"at-tab-bar__title\\">领取中心</div>undefined</div></div></div></div>"`;

exports[`AtTabBar Snap render initial AtTabBar 1`] = `"<div style=\\"background-color:;\\" class=\\"at-tab-bar\\"></div>"`;
